### PR TITLE
Pass secrets to approved workflow jobs

### LINKED ISSUE #251: Integration tests always fail for pull requests from a fork
### Description

For PRs sent from a fork, the integration tests always fail since secrets for the action are not being shared. Example in #250.

From the "Actions" tab in settings:

> Anyone with collaborator access to this repository can use these secrets and variables for actions. They are not passed to workflows that are triggered by a pull request from a fork.

It looks like there's [a workaround](https://michaelheap.com/access-secrets-from-forks/) that requires some additions to the workflow file to use secrets on a re-run. I'm unsure of other solutions, but I think this approach seems alright!

### What type of issue is this?
- [ ] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] example code related
- [x] testing related
- [x] discussion

### Requirements
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

### PR DESCRIPTION
**Pull request by zimeg**
###  Summary

This PR attempts to pass secrets to integration tests for PRs opened from a fork. Fixes #251.

PRs not opened by a `COLLABORATOR` will initially fail because secrets "are not passed to workflows that are triggered by a pull request from a fork" but checking out the `head.sha` on a re-run is [an apparent workaround for this](https://michaelheap.com/access-secrets-from-forks/) and workflows can be re-run by a maintainer!

### Notes

- Other options for `author_associations` include [`MEMBER`](https://docs.github.com/en/graphql/reference/enums#commentauthorassociation) which might work better for the check in `access_check`. I think a re-run will pass this check in any case though 🤔

### Requirements

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

**Comment by zimeg** [comment]
Good to know that `MEMBER` is different from `COLLABORATOR`! I'll try a re-run to see if that works :eyes:

failed workflow run

**Comment by zimeg** [comment]
Hmm... `MEMBER` passes the access check (unsure if this would be true for re-runs against other PRs) but secrets still aren't shared 🔐

**Comment by zimeg** [comment]
Trouble with secret passing might be a limitation of the `pull_request` event entirely! There's an option to use [`pull_request_target`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) instead which can access secrets but comes with many warnings around security. I think it's alright to use this event with caution and the "Require approval for outside collaborators" setting for Actions, but open to discussion!

**Review by seratch — APPROVED** [review]
Ideally, we wan to run tests for PRs but this workaround looks good to me at this time

**Comment by zimeg** [comment]
@seratch There are a few new changes so want to check that this looks good before merging.

The tests aren't running on this PR, but I believe this is because `pull_request` was removed and the updated `pull_request_target` doesn't yet exist on `main`.

Using `pull_request_target` for future PRs should start this workflow with necessary secrets and the `github.event.pull_request.head.sha` in `actions/checkout@v4` will use the changes in the PR when building and testing changes.

File: .github/workflows/main.yml
Comment (zimeg) [inline .github/workflows/main.yml:21]: The `pull_request_target` still uses the `event.pull_request` values from what I can tell: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

**Review by seratch — APPROVED** [review]
@zimeg Sorry for missing the additional changes here. All of them look good to me too. You can merge this PR tomorrow (or early next week).

**Comment by zimeg** [comment]
@seratch No worries at all! Thank you for another review 🙌

**Pull request merged**
